test/tests/materials/isoil/tests

```diff
@@ -67,6 +115,41 @@
     requirement = "The ComputeISoilStress class shall accurately construct the stress-strain curve using the GQ/H formulation without the use of the ISoilAction."
   [../]
 
+  [./GQH_AD]
+    # 1 element problem with ISoilAction and automatic differentiation to check auto-generated GQH stress-strain curves
+    type = Exodiff
+    input = HYS_GQH_AD.i
+    exodiff = HYS_GQH_out.e
+    cli_args = Outputs/file_base=HYS_GQH_out
+    abs_zero = 1e-9
+    prereq = GQH_noaction
+
+    requirement = "The ISoil action and ADComputeISoilStress class shall accurately model the nonlinear dynamic response of the system when automatic differentiation is used along with the auto-generated GQH stress-strain curves."
+  [../]
+
+  [./GQH_noaction_AD]
+    # 1 element problem without ISoilAction and automatic differentiation to check auto-generated GQH stress-strain curves
+    type = Exodiff
+    input = HYS_GQH_noaction_AD.i
+    exodiff = HYS_GQH_out.e
+    cli_args = Outputs/file_base=HYS_GQH_out
+    abs_zero = 1e-9
+    prereq = GQH_AD
+
+    requirement = "The ADComputeISoilStress class shall accurately model the nonlinear dynamic response of the system when automatic differentiation is used along with the auto-generated GQH stress-strain curves."
+  [../]
+
+  [./data_file_jac_check]
+    # 1 element problem with ISoilAction and automatic differentiation to check auto-generated GQH stress-strain curves
```

Comment:
What type of boundary conditions do you have for this test? If you had nodal boundary conditions getting applied to every element node, then all the jacobians would be manually set to 1, and your AD jacobians would not be getting tested. I'm guessing this is not the case, but I want to make sure. In general I usually use 2-3 elements so I'm sure I have at least one unconstrained Jacobian entry. Also do you know how many non-linear iterations this test takes? I usually like to get at least two because then that means I'm probably avoiding getting "lucky" in that I might have an initial condition where all variable gradients are zero or something like that.

Reply:
I am using two elements for this test and I am using DirichletBC on the bottom boundary of the bottom element and the top boundary of the top element. So the nodes that are shared between the two elements have free dofs. The iterations look like the one below on most timesteps and I am going through 8 timesteps in this test.

 0 Nonlinear |R| = 2.000501e-03
      0 Linear |R| = 2.000501e-03
      1 Linear |R| = -0.000000e+00
 1 Nonlinear |R| = 9.989995e-04
      0 Linear |R| = 9.989995e-04
      1 Linear |R| = 2.491007e-21
 2 Nonlinear |R| = 4.336809e-19
 Solve Converged!